test/test.tools.js

```diff
@@ -51,36 +61,53 @@ describe("Monkeypatching", () => {
 describe("Test tools for parsing new gmail body_params", () => {
     const gmail = new Gmail();
     const data =  JSON.parse(testData.new_gmail_archive_action_body_params);
+    const threads = gmail.tools.extract_from_graph(data, gmail.check.data.is_thread);
+    const threadData = threads.map(thread => gmail.tools.get_thread_data(thread))[0];
 
     it("get thread id", () => {
-        const thread = gmail.tools.get_thread_id(data);
+        const thread = gmail.tools.get_thread_id(threads[0]);
 
-        assert.equal(thread, 'thread-f:1600724307680265309');
+        assert.equal(thread, 'thread-f:1603171109786600032');
     });
+
     it("get thread data", () => {
-        const mockThreadData = data[2][7];
-        const threadData = gmail.tools.get_thread_data(data);
+        const mockThreadData = threads[0][2][7];
+        const threadData = gmail.tools.get_thread_data(threads[0]);
+        console.log(threadData);
 
         assert.deepEqual(threadData, mockThreadData);
     });
+
     it("get messages ids", () => {
-        const mockMessageIds = ['msg-f:1600724307680265309', 'msg-f:1600724938213937205', 'msg-f:1600725174437456906', 'msg-f:1600725319255992336', 'msg-f:1600725448529658711'];
-        const threadData = gmail.tools.get_thread_data(data);
+        const mockMessageIds = ['msg-f:1603171109786600032', 'msg-f:1603245801543734539', 'msg-f:1603245862071354412', 'msg-f:1603246018478443087', 'msg-f:1603256094012730022', 'msg-f:1603256564311088576', 'msg-f:1603256665279246114', 'msg-f:1603256682384715664', 'msg-f:1603376719891477511', 'msg-f:1603376909485932601', 'msg-f:1603376994923202634', 'msg-f:1603380395240179639'];
```

Comment:
I don't see why this test needed to be changed. Has there been changes to the test-data?

Also: What XHR test data are we actually working against here?

These changes are confusing me.

Reply:
Yes, I changed test-data, now we have clear data before I used prepared data